vital/types/metadata_traits.cxx

```diff
@@ -69,18 +70,34 @@ ::metadata_traits()
   KWIVER_VITAL_METADATA_TAGS( TABLE_ENTRY )
 
 #undef TABLE_ENTRY
+
+  // Create name table
+#define NAME_TABLE_ENTRY(TAG, NAME, TYPE, ...) \
+  m_name_trait_table[NAME] = trait_ptr(        \
+    static_cast< vital_meta_trait_base* >(new vital_meta_trait_object<VITAL_META_ ## TAG>() ) );
```

Comment:
This means that there are several instances of the trait, and that the pointer is not unique. Are you sure that's safe?

I think it might make more sense to rework this so that `TABLE_ENTRY` creates *one* pointer and has all the others point to that (note that this means the other tables are raw pointers, not smart pointers). Possibly the code would need to look like:
```c++
  {
    auto the_ptr = trait_ptr{ ... };
    m_name_trait_table[ NAME ] = the_ptr.get();
    m_enum_name_trait_table[ #TAG ] = the_ptr.get();
    m_trait_table[ VITAL_META_ ## TAG ] = std::move( the_ptr );
  }
```

In any case, I don't see the value of `TABLE_ENTRY`, `NAME_TABLE_ENTRY` and `ENUM_NAME_TABLE_ENTRY` being separate.

Reply:
Addressed with Leotta's method.